cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1995,6 +1995,25 @@ func (p *BicepProvider) ensureParameters(
 			}
 		}
 
+		if param.IsAutoGen() {
+			// IsAutoGen already checked the azdMetadata exists
+			azdMetadata, _ := param.AzdMetadata()
+			genValue, err := autoGenerate(key, azdMetadata)
```

Comment:
Just wondering how this relates to setting in the config. In practice, this ends up always generating a value regardless?

Reply:
Looks like it's checked up a few lines.